### PR TITLE
#94 : Add namespace notion to ubiquity_<install,uninstall,deployments>.sh

### PR DESCRIPTION
**Pull request by shay-berman**
All details inside #94

---
This change is [Reviewable](https://reviewable.io/reviews/ibm/ubiquity-k8s/95)

### CODE REVIEW

File: scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh [hunk 1]
```diff
@@ -48,16 +48,18 @@
...
-   echo "       status : kubectl get to all the ubiquity components"
-   echo "       getall : kubectl get configmap,storageclass,pv,pvc,service,daemonset,deployment,pod"
-   echo "       getallwide : getall but with -o wide"
-   echo "       collect_logs : Create a directory with all Ubiquity logs"
-   echo "       sanity    : create pvc and pod and then delete them"
-   echo "       help      : Show this usage"
+   echo "Usage, $0 -a [`echo $actions | sed 's/ /|/g'`] [-n <namespace>] [-h]"
+   echo " -a <action>"
+   echo "   start  : Create ubiquity, provisioner deployments, flex daemonset and ubiquity-db deployments"
+   echo "   stop   : Delete ubiquity-db(wait for deletion), provisioner deployment, flex daemonset, ubiquity deployment"
```
Comment (alonm) [inline scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh]: missing space

File: scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh [hunk 2]
```diff
@@ -48,16 +48,18 @@
...
+   echo "   status : kubectl get to all the ubiquity components"
+   echo "   getall : kubectl get configmap,storageclass,pv,pvc,service,daemonset,deployment,pod"
```
Comment (alonm) [inline scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh]: 'kubectl get configmap,storageclass,pv,pvc,service,daemonset,deployment,pod' => 'display  configmap,storageclass,pv,pvc,service,daemonset,deployment,pod components'

File: scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh [hunk 3]
```diff
@@ -69,27 +71,30 @@ function help()
 function start()
 {
     echo "Make sure ${UBIQUITY_DB_PVC_NAME} exist and bounded to PV (if not exit), before starting."
```
Comment (alonm) [inline scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh]: exist => exists

File: scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh [hunk 4]
```diff
@@ -69,27 +71,30 @@ function help()
...
-    kubectl create -f ${YML_DIR}/ubiquity-db-deployment.yml
-    echo "Finished to start ubiquity components. Run $0 status to get more details."
+    wait_for_item pvc ${UBIQUITY_DB_PVC_NAME} ${PVC_GOOD_STATUS} 5 2 $NS
+
+    kubectl create $nsf -f ${YML_DIR}/ubiquity-deployment.yml
+    kubectl create $nsf -f ${YML_DIR}/ubiquity-k8s-provisioner-deployment.yml
+    kubectl create $nsf -f ${YML_DIR}/ubiquity-k8s-flex-daemonset.yml
+    kubectl create $nsf -f ${YML_DIR}/ubiquity-db-deployment.yml
+    # TODO add some wait for deployment before run the ubiquity-db
+    echo "Finished to start ubiquity components. Note : View ubiquity status by : $> $0 -a status -n $NS"
```
Comment (alonm) [inline scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh]: 'Finished to start ubiquity components' => 'Initiated ubiquity components'

File: scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh [hunk 5]
```diff
@@ -69,27 +71,30 @@ function help()
...
+    wait_for_item_to_delete pod "ubiquity-db-" 10 4 regex $NS  # to match the prefix of the pod
 
-    kubectl delete -f $YML_DIR/ubiquity-k8s-provisioner-deployment.yml
-    kubectl delete -f ${YML_DIR}/ubiquity-k8s-flex-daemonset.yml
-    kubectl delete -f $YML_DIR/ubiquity-deployment.yml
-    echo "Finished to stop ubiquity deployments. Run $0 status to get more details."
+    $kubectl_delete -f $YML_DIR/ubiquity-k8s-provisioner-deployment.yml
+    $kubectl_delete -f ${YML_DIR}/ubiquity-k8s-flex-daemonset.yml
+    $kubectl_delete -f $YML_DIR/ubiquity-deployment.yml
+    echo "Finished to stop ubiquity deployments.   Note : View ubiquity status by : $> $0 -a status -n $NS"
```
Comment (alonm) [inline scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh]: 'Finished to stop ubiquity deployments' => 'Stopping ubiquity components'

**Comment by ranhrl** [comment]
Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions.

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 55 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-55:-Kw01eSYoetfkfkCvwTD:b6kkudj) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L55)):*
<details><summary><i>Previously, alonm wrote…</i></summary><blockquote>

'kubectl .. components' =>'display all ubiquity components'
</blockquote></details>

I agree, Shay, did you ask for Dima's review for all your user interactive experience?

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 56 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-56:-Kw01t1bPYYeQSc1x8l4:bifi7gs) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L56)):*
<details><summary><i>Previously, alonm wrote…</i></summary><blockquote>

'kubectl get configmap,storageclass,pv,pvc,service,daemonset,deployment,pod' => 'display  configmap,storageclass,pv,pvc,service,daemonset,deployment,pod components'
</blockquote></details>

If this is to be used by users, it should explain itself better, like "display all related deployed entities" and then have a title for each entity type.

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 78 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw02OzRuUHlSjeCdSLS:-Kw02OzRuUHlSjeCdSLT:b8a4hdp) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L78)):*
> ```Shell
>     kubectl create $nsf -f ${YML_DIR}/ubiquity-deployment.yml
>     kubectl create $nsf -f ${YML_DIR}/ubiquity-k8s-provisioner-deployment.yml
>     kubectl create $nsf -f ${YML_DIR}/ubiquity-k8s-flex-daemonset.yml
> ```

You are not checking for any return codes on any of your commands. You should have an executer function to check the return status and exit nicely on errors. A more advanced feature would be to pick up where last action failed (be idempotent) - TBD.

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 149 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw02wWf5rXXEnmW6Oty:-Kw02wWf5rXXEnmW6Otz:b4cmm7r) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L149)):*
> ```Shell
>     cmd="kubectl get $nsf cm/k8s-config pv/ibm-ubiquity-db pvc/ibm-ubiquity-db svc/ubiquity svc/ubiquity-db  daemonset/ubiquity-k8s-flex deploy/ubiquity deploy/ubiquity-db deploy/ubiquity-k8s-provisioner"
>     echo $cmd
>     echo '---------------------------------------------------------------------'
> ```

This should be defined in a constant DASHES or something.

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 286 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw03HmmQ98gm0i43xGn:-Kw03HmmQ98gm0i43xGo:b-x4dxnw) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L286)):*
> ```Shell
>   esac
> done
> echo "Working in namespace [$NS]."
> ```

Deployment namespace is [$NS]

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95)*

**Comment by ranhrl** [comment]
Review status: 0 of 7 files reviewed at latest revision, 10 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw04NAmNNlvwBPRjEmn:-Kw04NAmNNlvwBPRjEmo:bqqa7f9) (no related file):*
Are you sure it won't be easier and nicer to deploy using python with https://github.com/kubernetes-incubator/client-python/tree/master/kubernetes

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95)*

**Comment by midoblgsm** [comment]
*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 73 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw1Hq22jC1wHydSkYd6:-Kw1Hq23GZkDSVfI-Wbw:b-2ps7dc) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L73)):*
> ```Shell
> function start()
> {
>     echo "Make sure ${UBIQUITY_DB_PVC_NAME} exist and bounded to PV (if not exit), before starting."
> ```

Make sure ${UBIQUITY_DB_PVC_NAME} exists and bound to PV (exit otherwise)...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95)*

**Comment by midoblgsm** [comment]
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95)*

**Comment by shay-berman** [comment]
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 54 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-54:-Kw6gO-iZGNXvk7ocu_P:bsnxja6) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L54)):*
<details><summary><i>Previously, alonm wrote…</i></summary><blockquote>

missing space
</blockquote></details>

I am not sure where is the missing space, maybe after the (). please be more specific

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 55 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-55:-Kw6gn024nSA3xiqi-2Q:bjqb2sj) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L55)):*
> display all ubiquity components
fixes
Dima should review it soon, there is open ticket for it.

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 56 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-56:-Kw6hWYZ6m9IJhqZTHz6:b-axpioq) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L56)):*
<details><summary><i>Previously, ranhrl (Ran Harel) wrote…</i></summary><blockquote>

If this is to be used by users, it should explain itself better, like "display all related deployed entities" and then have a title for each entity type.
</blockquote></details>

improved a bit 
anyway user that run this command understand the kubectl get syntax

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 73 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw1Hq22jC1wHydSkYd6:-Kw6hntL0gQrBCSHbGAe:bk13xrm) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L73)):*
<details><summary><i>Previously, midoblgsm (Mohamed Mohamed) wrote…</i></summary><blockquote>

Make sure ${UBIQUITY_DB_PVC_NAME} exists and bound to PV (exit otherwise)...
</blockquote></details>

applied your text

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 78 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw02OzRuUHlSjeCdSLS:-Kw6iNpXF3WTW7Z1VcpW:bhm3km5) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L78)):*
<details><summary><i>Previously, ranhrl (Ran Harel) wrote…</i></summary><blockquote>

You are not checking for any return codes on any of your commands. You should have an executer function to check the return status and exit nicely on errors. A more advanced feature would be to pick up where last action failed (be idempotent) - TBD.
</blockquote></details>

set -e  should be good enough for these kubectl create
if there is an issue, user will see the relevant error that kubectl returns

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 81 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-81:-Kw6is4iHM1cKN5fsR0j:b-71qr2p) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L81)):*
> Initiated ubiquity components
will consult with dima

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 97 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kvwl1l2KUD_ll6lcftW-r1-97:-Kw6iydfcl67o580SdO8:b-elgm5s) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L97)):*
<details><summary><i>Previously, alonm wrote…</i></summary><blockquote>

'Finished to stop ubiquity deployments' => 'Stopping ubiquity components' 
</blockquote></details>

stopping... why, its the end of the operation. anyway will consult with Dima

---

*[scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh, line 149 at r1](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95#-Kw02wWf5rXXEnmW6Oty:-Kw6k90GPzUUeO2atyh3:b-7ca9hy) ([raw file](https://github.com/ibm/ubiquity-k8s/blob/bc6d60b275a7155838873e583e5e922a9fc01b33/scripts/k8s_scbe_all_in_one/ubiquity_deployments.sh#L149)):*
> DASHES
there any tricky ways to do it, but i prefer to stay simple

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-k8s/95)*

**Pull request merged**